src/targets/gpu/gemm_impl.cpp

```diff
@@ -593,27 +621,36 @@ int32_t gemm_finalize(context& ctx,
 
     if(solution_idx == 0)
     {
-        auto gemm_item = gemm_impl<float>(output_shape, input_shapes, alpha, beta, compute_fp32);
+        auto gemm_item = gemm_impl<T>(output_shape, input_shapes, alpha, beta, compute_fp32);
         solution_idx   = gemm_item.tune(ctx, input_shapes);
     }
     else
     {
         // If a tuned solution index is already given, don't tune again but validate
         // in case the data was tuned with a different rocBLAS version
-        auto gemm_item = gemm_impl<float>(output_shape, input_shapes, alpha, beta, compute_fp32);
+        auto gemm_item = gemm_impl<T>(output_shape, input_shapes, alpha, beta, compute_fp32);
         solution_idx   = gemm_item.validate(ctx, input_shapes, solution_idx);
     }
+    gemm_save_solution(ctx, output_shape, input_shapes, solution_idx);
```

Comment:
Saving should be after tuning inside `if` clause.

Reply:
Done